app/lib/three_scale/bot_protection/form.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+module ThreeScale
+  module BotProtection
+    module Form
+      include Recaptcha::Adapters::ViewMethods
+
+      private
+
+      def bot_protection_inputs
+        return ''.html_safe unless bot_protection_enabled?
+
+        recaptcha_v3(action: template.controller.controller_path)
+      end
+
+      def bot_protection_level
+        template.site_account.settings.spam_protection_level
+      end
+
+      def bot_protection_enabled?
+        Recaptcha.captcha_configured? && !template.logged_in? && bot_protection_level != :none
```

Comment:
You are right. I was looking for a way to unify `bot_protection_enabled?`, but it's not straightforward because forms and controllers have different attributes, and the resulting code wouldn't be good, IMO.

Finally I pushed https://github.com/3scale/porta/pull/3645/commits/f43c44d5db39c1f382972370efa1e40f079c81a5. Which checks for `Recaptcha.captcha_configured?`, simplifies `bot_check` and at least unifies the same behavior in both sides, tough it doesn't unify the code.

Reply:
It seems like that `#bot_protection_enabled?` can be in a module included by both classes, no?

Reply:
I did it in https://github.com/3scale/porta/pull/3645/commits/df9096e51702ecd5b4c8fe68b37c7823afc75b93